pkg/virtctl/memorydump/memorydump.go

```diff
@@ -0,0 +1,367 @@
+	cmd.Flags().StringVar(&accessMode, accessModeArg, "", "The access mode for the PVC.")
+	cmd.Flags().StringVar(&outputFile, "output", "", "Specifies the output path of the memory dump to be downloaded.")
+
+	return cmd
+}
+
+func (c *command) run(args []string) error {
+	namespace, _, err := c.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+	}
+
+	vmName := args[1]
+	switch args[0] {
+	case "get":
+		return getMemoryDump(namespace, vmName, virtClient)
+	case "download":
+		return downloadMemoryDump(namespace, vmName, virtClient)
+	case "remove":
+		return removeMemoryDump(namespace, vmName, virtClient)
+	default:
+		return fmt.Errorf("invalid action type %s", args[0])
+	}
+}
+
+func calcMemoryDumpExpectedSize(vmName, namespace string, virtClient kubecli.KubevirtClient) (*resource.Quantity, error) {
+	vmi, err := virtClient.VirtualMachineInstance(namespace).Get(vmName, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return kutil.CalcExpectedMemoryDumpSize(vmi), nil
+}
+
+func calcPVCNeededSize(memoryDumpExpectedSize *resource.Quantity, storageClass *string, virtClient kubecli.KubevirtClient) (*resource.Quantity, error) {
+	cdiConfig, err := virtClient.CdiClient().CdiV1beta1().CDIConfigs().Get(context.Background(), configName, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		// can't properly determine the overhead - continue with default overhead of 5.5%
+		fmt.Println(fsOverheadMsg)
+		return storagetypes.GetSizeIncludingGivenOverhead(memoryDumpExpectedSize, filesystemOverhead)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	fsVolumeMode := k8sv1.PersistentVolumeFilesystem
+	if *storageClass == "" {
+		storageClass = nil
+	}
+
+	return storagetypes.GetSizeIncludingFSOverhead(memoryDumpExpectedSize, storageClass, &fsVolumeMode, cdiConfig)
+}
+
+func generatePVC(size *resource.Quantity, claimName, namespace, storageClass, accessMode string) (*k8sv1.PersistentVolumeClaim, error) {
+	pvc := &k8sv1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      claimName,
+			Namespace: namespace,
+		},
+		Spec: k8sv1.PersistentVolumeClaimSpec{
+			Resources: k8sv1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceStorage: *size,
+				},
+			},
+		},
+	}
+
+	if storageClass != "" {
+		pvc.Spec.StorageClassName = &storageClass
+	}
+
+	if accessMode == string(k8sv1.ReadOnlyMany) {
+		return nil, fmt.Errorf("cannot dump memory to a readonly pvc, use either ReadWriteOnce or ReadWriteMany if supported")
+	}
+	if accessMode != "" && accessMode != string(k8sv1.ReadWriteOnce) && accessMode != string(k8sv1.ReadWriteMany) {
+		return nil, fmt.Errorf("invalid access mode, use either ReadWriteOnce or ReadWriteMany if supported")
+	}
+	if accessMode != "" {
```

Comment:
Maybe is cleaner to include the two previous `ifs` inside this block.

Reply:
ok